R/content.R

```diff
@@ -223,7 +236,10 @@ ContentToolResult <- new_class(
           "."
         )
       }
-    )
+    ),
+    extra = class_list,
+    id = prop_string(allow_null = TRUE),
+    request = NULL | ContentToolRequest
```

Comment:
If feels a bit strange that you're allowed to pass in values for these fields, even though they'll never be used. What if we instead had a `tool_result()`/`ToolResult` that gets upgraded to `ContentToolResult`?

Reply:
I considered this but it would defeat the purpose: If you return a `ToolResult`, you can't control the class of the object stored in the turns. On the other hand, if your custom class has `parent = ContentToolResult`, we fill in the details and return your subclassed object.

Reply:
As I mentioned above, though, we could get it down to just the `@request` property that gets filled in.

Reply:
Ahh right, makes sense 👍